common/interactivetx.c

```
@@ -0,0 +1,687 @@
+		case WIRE_TX_ADD_INPUT:
+		case WIRE_TX_REMOVE_INPUT:
+		case WIRE_TX_ADD_OUTPUT:
+		case WIRE_TX_REMOVE_OUTPUT:
+		case WIRE_TX_COMPLETE:
+			return msg;
+		case WIRE_TX_SIGNATURES:
+		case WIRE_FUNDING_LOCKED:
+		case WIRE_INIT_RBF:
+		case WIRE_OPEN_CHANNEL2:
+		case WIRE_INIT:
+		case WIRE_ERROR:
+		case WIRE_OPEN_CHANNEL:
+		case WIRE_ACCEPT_CHANNEL:
+		case WIRE_FUNDING_CREATED:
+		case WIRE_FUNDING_SIGNED:
+		case WIRE_CLOSING_SIGNED:
+		case WIRE_UPDATE_ADD_HTLC:
+		case WIRE_UPDATE_FULFILL_HTLC:
+		case WIRE_UPDATE_FAIL_HTLC:
+		case WIRE_UPDATE_FAIL_MALFORMED_HTLC:
+		case WIRE_COMMITMENT_SIGNED:
+		case WIRE_REVOKE_AND_ACK:
+		case WIRE_UPDATE_FEE:
+		case WIRE_UPDATE_BLOCKHEIGHT:
+		case WIRE_CHANNEL_REESTABLISH:
+		case WIRE_ANNOUNCEMENT_SIGNATURES:
+		case WIRE_GOSSIP_TIMESTAMP_FILTER:
+		case WIRE_OBS2_ONION_MESSAGE:
+		case WIRE_ONION_MESSAGE:
+		case WIRE_ACCEPT_CHANNEL2:
+		case WIRE_ACK_RBF:
+		case WIRE_CHANNEL_ANNOUNCEMENT:
+		case WIRE_CHANNEL_UPDATE:
+		case WIRE_NODE_ANNOUNCEMENT:
+		case WIRE_QUERY_CHANNEL_RANGE:
+		case WIRE_REPLY_CHANNEL_RANGE:
+		case WIRE_QUERY_SHORT_CHANNEL_IDS:
+		case WIRE_REPLY_SHORT_CHANNEL_IDS_END:
+		case WIRE_WARNING:
+		case WIRE_PING:
+		case WIRE_PONG:
+		case WIRE_SHUTDOWN:
+		case WIRE_SPLICE:
+		case WIRE_SPLICE_ACK:
+#if EXPERIMENTAL_FEATURES
+		case WIRE_STFU:
+#endif
+		*error = tal_fmt(ctx,
+				 "Received invalid message from peer: %d", t);
+		return NULL;
+		}
+	}
+}
+
+static char *send_next(const tal_t *ctx,
+		       struct interactivetx_context *ictx,
+		       bool *finished)
+{
+	struct channel_id *cid = &ictx->channel_id;
+	struct psbt_changeset *set = ictx->change_set;
+	u64 serial_id;
+	u8 *msg = NULL;
+
+	*finished = false;
+
+	if(!set)
+		goto tx_complete;
+
+	if(tal_count(set->added_ins) != 0) {
+
+		const struct input_set *in = &set->added_ins[0];
+		struct bitcoin_outpoint outpoint;
+		u8 *prevtx;
+
+		if (!psbt_get_serial_id(&in->input.unknowns, &serial_id))
+			abort();
```

can't decide if a hard fail is ok or if we should return an error? it's not recoverable, but maybe crashing isn't the best policy?